src/libServer/Server.cpp

```diff
@@ -132,47 +143,80 @@ string Server::createTransaction(const Json::Value& _json)
    	
 }
 
-Json::Value Server::getTransaction(const string & transactionHash)
+Json::Value Server::GetTransaction(const string & transactionHash)
 {
-	return "Hello";
-}
+	LOG_MARKER();
+	TxBodySharedPtr tx;
+	TxnHash tranHash(transactionHash);
+	bool isPresent = BlockStorage::GetBlockStorage().GetTxBody(tranHash, tx);
+	if(!isPresent)
+	{
+		Json::Value _json;
+		_json["error"] = "Txn Hash not Present";
+		return _json;
+	}
+	Transaction txn(tx->GetVersion(), tx->GetNonce(),tx->GetToAddr(), tx->GetSenderPubKey(), tx->GetAmount(), tx->GetSignature());
+	return JSONConversion::convertTxtoJson(txn);
+ }
 
-Json::Value Server::getDsBlock(const string & blockHash)
+Json::Value Server::GetDsBlock(const string & blockNum)
 {
-	return "Hello";
+
+	
+	try
+	{
+		boost::multiprecision::uint256_t BlockNum(blockNum);
+		return JSONConversion::convertDSblocktoJson(m_mediator.m_dsBlockChain.GetBlock(BlockNum));
+	}
+	catch(const char* msg)
+	{
+		Json::Value _json;
+		_json["Error"] = msg;
+		return _json;
+	}
 }
 
-Json::Value Server::getTxBlock(const string & blockHash)
+Json::Value Server::GetTxBlock(const string & blockNum)
 {
-	return "Hello";
+	try
+	{
+		boost::multiprecision::uint256_t BlockNum(blockNum);
+		return JSONConversion::convertTxBlocktoJson(m_mediator.m_txBlockChain.GetBlock(BlockNum));
+	}
+	catch(const char* msg)
```

Comment:
Is this a recommended try-catch usage from jsoncpp? I'm a bit concern about who allocates the error string and this may cause memory leak if no one frees it

Reply:
The exception originates from GetBlock() function